### PR TITLE
Fix crash caused by queuing multiple pick up orders for a single unit.

### PR DESCRIPTION
**Pull request by darkademic**
Feel free to suggest a simpler check, or a better location for it.

If you queue multiple pick up orders on a single unit, the game will crash once the Carryall has picked up the unit and has fully taken off again (as it tries to perform another PickupUnit activity).

This fixes it.

### CODE REVIEW

**Comment by PunkPun** [comment]
It crashes with null exception because it failed to reserve in `OnFirstRun` and now it doesn't have child activities. To fix this we should make sure we cancel the activity on failing to reserve. I recon a flag should be good as we can't call `Cancel` in `OnFirstTick`

**Review by PunkPun — COMMENTED** [review]
It crashes with null exception because it failed to reserve in `OnFirstRun` and now it doesn't have child activities. To fix this we should make sure we cancel the activity on failing to reserve. I recon a flag should be good as we can't call `Cancel` in `OnFirstTick`

**Comment by darkademic** [comment]
> It crashes with null exception because it failed to reserve in `OnFirstRun` and now it doesn't have child activities. To fix this we should make sure we cancel the activity on failing to reserve. I recon a flag should be good as we can't call `Cancel` in `OnFirstTick`

Aha, something like this (see updated code)? That also fixes it.

File: OpenRA.Mods.Common/Activities/PickupUnit.cs [hunk 1]
```diff
@@ -61,11 +62,13 @@ protected override void OnFirstRun(Actor self)
 				QueueChild(new Fly(self, Target.FromActor(cargo)));
 				QueueChild(new FlyIdle(self, idleTurn: false));
 			}
+			else
+				reserveFailed = true;
```
Comment (PunkPun) [inline OpenRA.Mods.Common/Activities/PickupUnit.cs:69]: reserve fails is the above `return` as well, tough as checks match it is caught in Tick. There shouldn't be any harm in setting `reserveFailed = true` in the above `return`

Comment (darkademic) [reply OpenRA.Mods.Common/Activities/PickupUnit.cs:69]: Yea makes sense.

**Review by PunkPun — APPROVED** [review]

**Comment by PunkPun** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/b81057949dfead7ae125c46772107a500836ddb7...46637308e9b4236797cc82d0494966551045fc8c)

**Pull request merged**